### ISSUE TITLE
Users can't edit wiki

### ISSUE THREAD

**Issue by CendioOssman** [novnc/noVNC#1922]
**Describe the bug**
Only maintainers can currently edit our wiki, making it almost useless as the point of the wiki is to have documentation that the community can maintain.

**Additional context**
The reason the wiki is locked down is because there is too much malware being added to GitHub wikis. And GitHub provides absolutely no tools to deal with the issue.

In other words, getting GitHub doesn't seem feasible. We need to look at alternatives.

This could be an option where we can still host things on GitHub:

https://github.com/Drassil/git-wiki-theme

**Comment by pallandir** [comment]
Hi. Hope you're doing well.

Why not consider solutions like Astro.js and Starlight for documentation?  (https://starlight.astro.build/)  It can also be deployed to GitHub Pages (https://docs.astro.build/en/guides/deploy/github/). Let me know if I can help solve this issue.

**Comment by CendioOssman** [comment]
That does not seem to be a wiki where users can contribute?

**Comment by pallandir** [comment]
Yes, you're right. It doesn't provide a direct way to collaborate online, and you would need pull requests as a normal project in order to accept documentation contributions. However, I thought about them because you would have better control (related to the malware issues of wikis) over the submissions, and they are still easy to deploy and setup.

**Comment by CendioOssman** [comment]
A pull request based model could definitely be an option. I guess we'll need to set up one of these things and play around with it.

**Comment by pallandir** [comment]
## Starlight Demo  

I've built a quick demo showcasing how Starlight documentation could look like! You can check it out here:  

🔗 [Documentation demo](https://stackblitz.com/~/github.com/pallandir/documentation-website)  

### Considerations

 - Documentation under src/content/docs/documentation is autogenerated. (As long as you create a folder or file here it will appear in the sidebar once running the project) 
- Landing page is the default Hero but it can be anything. 
- Contributing to this project can be editing documetation under src/content/docs or updating the site itself (configuration, UI ...)
- Documentation can be written as Markdown or mdx : [more on this here](https://starlight.astro.build/components/using-components/#using-a-component-in-mdx)  
- Each documentation file has to contain default title (only field mandatory) and can be customized thanks to frontmatter.

### Setup and deployment

This project can be added to an already existing repository under a subfolder (e.g /docs) or into a new one to split actual implementation of NoVnc and its documentation. 

Deployment can be done with GitHub pages or any other host: [deploy astro project](https://docs.astro.build/en/guides/deploy/)

### final words

If you feel this solution can be relevant for your use-case I would be open and interested to contribute to its implementation / configuration / documentation migration.
